lib/messenger/index.js

```diff
@@ -586,10 +587,13 @@ function _callSubscriber(subscriber, message, data, callback, _synchronous) {
             subscriber.options.dispatchTimes--;
     }
 
-    if (synchro)
+    if (synchro) {
         subscriber.subscriber.call(subscriber.context, message, data, callback);
-    else
-        _setTimeout(function() { subscriber.subscriber.call(subscriber.context, message, data, callback); }, 0);
+    } else {
+        _setTimeout(function() {
+            if (!this._destroyed) subscriber.subscriber.call(subscriber.context, message, data, callback);
+        }.bind(this), 0);
```

Comment:
binding is wasteful, better to use scope variable (`var self = this` - that's how it's done everywhere)